subprojects/gst-plugins-good/gst/rtp/gstrtph266pay.c

```diff
@@ -0,0 +1,754 @@
+    default:
+      break;
+  }
+
+  if (ret != GST_FLOW_OK)
+    return FALSE;
+
+  return GST_RTP_BASE_PAYLOAD_CLASS (parent_class)->sink_event (rtpbasepay,
+      event);
+}
+
+static void
+_process_nalu (GstRtpH266Pay * rtph266pay, GstBuffer * nalu_buf)
+{
+  Nalu *nalu;
+
+  g_assert (nalu_buf);
+
+  // Parse NALU
+  nalu = _nalu_new (nalu_buf);
+  if (!nalu) {
+    GST_WARNING_OBJECT (rtph266pay,
+        "Couldn't decode NALU %" GST_PTR_FORMAT ". Dropping it.", nalu_buf);
+    return;
+  }
+  GST_DEBUG_OBJECT (rtph266pay, "NALU decoded: %" NALU_PTR_FORMAT,
+      NALU_ARGS (nalu));
+
+  _update_xps_cache (rtph266pay, nalu);
+  g_queue_push_tail (&rtph266pay->nalus, nalu);
+}
+
+static void
+_update_xps_cache (GstRtpH266Pay * rtph266pay, const Nalu * nalu)
+{
+  GstRTPBasePayload *rtpbasepay = GST_RTP_BASE_PAYLOAD (rtph266pay);
+  GHashTable *map = rtph266pay->xps_id_nalu_map;
+  guint8 type = nalu->type;
+  guint8 id = nalu->xps_id;
+  GstClockTime pts;
+  Nalu *new_nalu;
+  gpointer *key;
+
+  if (!NALU_IS_PARAMETER_SET (nalu))
+    return;
+
+  new_nalu = _nalu_copy (nalu);
+  new_nalu->au_start = FALSE;
+  new_nalu->au_end = FALSE;
+  key = GINT_TO_POINTER ((type << 8) | id);
+  if (g_hash_table_insert (map, key, new_nalu))
+    GST_DEBUG_OBJECT (rtph266pay, "XPS(%u,%u) cached", type, id);
+  else
+    GST_DEBUG_OBJECT (rtph266pay, "XPS(%u,%u) replaced", type, id);
+
+  pts = GST_BUFFER_PTS (nalu->nalu_buf);
+  if (GST_CLOCK_TIME_IS_VALID (pts)) {
+    rtph266pay->last_xps_recv_ts =
+        gst_segment_to_running_time (&rtpbasepay->segment, GST_FORMAT_TIME,
+        pts);
+  }
+}
+
+static void
+_clear_xps_cache (GstRtpH266Pay * rtph266pay)
+{
+  rtph266pay->last_xps_recv_ts = GST_CLOCK_TIME_NONE;
+  g_hash_table_remove_all (rtph266pay->xps_id_nalu_map);
+}
+
+static void
+_queue_xps_cache (GstRtpH266Pay * rtph266pay)
+{
+  GstRTPBasePayload *rtpbasepay = GST_RTP_BASE_PAYLOAD (rtph266pay);
+  GHashTableIter iter;
+  GstClockTime pts;
+  Nalu *nalu;
+
+  g_hash_table_iter_init (&iter, rtph266pay->xps_id_nalu_map);
+  while (g_hash_table_iter_next (&iter, NULL, (gpointer) & nalu)) {
+    g_queue_push_head (&rtph266pay->nalus, _nalu_copy (nalu));
+
+    pts = GST_BUFFER_PTS (nalu->nalu_buf);
+    if (GST_CLOCK_TIME_IS_VALID (pts)) {
+      rtph266pay->last_xps_sent_ts =
+          gst_segment_to_running_time (&rtpbasepay->segment, GST_FORMAT_TIME,
+          pts);
+    }
+  }
+}
+
+static gboolean
+_has_to_send_xps (GstRtpH266Pay * rtph266pay, const Nalu * nalu)
+{
+  // TODO
+  // if nalu_type is not one of the allowed (is a non-VCL maybe?)
+  //  return FALSE;
+  //
+  // if (rtph266pay->last_xps_sent_ts < 0) && NALU_IS_IDR (nalu))
+  //  return TRUE;
+  //
+  // GstClockTime running_time =
+  //   gst_segment_to_running_time (&rtpbasepay->segment,
+  //       GST_FORMAT_TIME, pts);
+  //
+  // if (rtph266pay->last_xps_sent_ts + (rtph266pay->config_interval * GST_SECOND) >= running_time)
+  //   return TRUE;
+
+  return FALSE;
```

Comment:
If the xps cache is not used, please delete the code. Otherwise, comment all the code related with `#if 0`.

Reply:
Done. I've moved everything related to config-interval into https://github.com/fluendo/gstreamer/pull/42